peer_mentoring/account_management/views.py

```diff
@@ -111,10 +111,16 @@ def send_friend_request(request, user_id):
 def accept_friend_request(request, request_id):
     friend_request = get_object_or_404(FriendRequest, id=request_id)
     if friend_request.receiver == request.user.userprofile:
-        request.user.userprofile.friends.add(
+        friend_request.receiver.friends.add(
             friend_request.sender
         )  # assumes 'friends' is a ManyToManyField on User
-        friend_request.delete()
+        friend_request.sender.friends.add(
```

Comment:
to add friend on both sides (although I wonder if a separate table, similar to FriendRequest, would have been better, but let's leave it for now, as long as it works ...)